test/TestEncodingProtocol.hs

```diff
@@ -0,0 +1,31 @@
+{-# LANGUAGE OverloadedStrings #-}
+module TestEncodingProtocol where
+import qualified Data.ByteString as B
+import Control.Concurrent.Chan.Unagi as U
+import Control.Concurrent.MVar (newEmptyMVar, takeMVar)
+import Test.HUnit (assertEqual)
+import Test.Framework.Providers.HUnit (testCase)
+
+import Clapi.Types (Message(MsgError))
+import Clapi.Protocol (runProtocolIO, (<->), sendRev, sendFwd, waitThen)
+import Clapi.EncodingProtocol (serialiser)
+
+tests = [
+    testCase "packetised round trip" testPacketisedRoundTrip
+    ]
+
+testPacketisedRoundTrip = do
+    (bi, bo) <- U.newChan
+    (mi, mo) <- U.newChan
+    mv <- newEmptyMVar
```

Comment:
I'm not sure I follow what this `MVar` is for. Could you do any indefinitely blocking operation where it's used on the next line?

Reply:
I think so yes, just didn't know how to do it any other way.